PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+    prompt = no
+    req_extensions = v3_req
+    
+    [ v3_req ]
+    basicConstraints = CA:FALSE
+    keyUsage = digitalSignature, keyEncipherment, dataEncipherment, nonRepudiation
+    extendedKeyUsage = serverAuth, clientAuth
+    subjectAltName = @alternate_names
+
+    [ req_distinguished_name ]
+    commonName = ' +$commonName+ '
+    countryName = ' +$country+ '
+    stateOrProvinceName = ' +$state+ '
+    localityName = ' +$locality+ '
+    0.organizationName = ' +$organization+ '
+    organizationalUnitName = ' +$organizationUnit+ '
+    
+    [ alternate_names ]
+    '
+    # Parsing SAN string
+    $sanArrary = $subjectAlternativeNames -split "\s*,\s*"
+    $sanCount = 1
+
+    foreach($san in $sanArrary) {
+        $cfgString = $cfgString + "DNS." + $sanCount + " = " + $san + "`n"
+        $sanCount++
+    }
+
+    # Generate RSA private key and CSR file
+    $cfgString | Out-File $cfgFilePath
+    openssl req -new -sha256 -nodes -out $csrFilePath -keyout $orginalKeyFilePath -config $cfgFilePath >> $null 2>$null
+    if ($opensslVersion -match "OpenSSL 3.") {
+        openssl rsa -in $orginalKeyFilePath -out $keyFilePath -traditional >> $null 2>$null
+    } else {
+        openssl rsa -in $orginalKeyFilePath -out $keyFilePath >> $null 2>$null
+    }
+
+    # Clean up
+    remove-item $cfgFilePath
+    remove-item $orginalKeyFilePath
+    
+}
+Export-ModuleMember -Function Invoke-GeneratePrivateKeyAndCsrFiles
+
+Function Invoke-RequestSignedCertificateByMsca {
+    <#
+        .SYNOPSIS
+        Request CSR to be signed by a MSCA Certificate Authority.
+
+        .DESCRIPTION
+        The Invoke-RequestSignedCertificateByMsca cmdlet connects to MSCA web enrollment to sign a CSR.  
+        
+        .EXAMPLE
+        Invoke-RequestSignedCertificateByMsca -caFqdn "rpl-ad01.rainpole.io" -username "Administrator" -password "VMw@re1!" -csrFilePath
+         "C:\certificates\xint-vrli01.rainpole.io.csr" -certDir "C:\certificates" -certificateTemplate "VMware"
+        This example will request the xint-vrli01 CSR file to be signed by Certificate Authority rpl-ad01.rainpole.io.
+
+        .EXAMPLE
+        Invoke-RequestSignedCertificateByMsca -caFqdn "rpl-ad01.rainpole.io" -username "Administrator" -password "VMw@re1!" -csrFilePath
+         "C:\certificates\xint-vrli01.rainpole.io.csr" -certDir "C:\certificates" -certificateTemplate "VMware" -getCArootCert
+        This example will request the xint-vrli01 CSR file to be signed by Certificate Authority rpl-ad01.rainpole.io and also will retrieve 
+        the Certificate Authority's root chain certificate.  
+
+        .PARAMETER caFqdn
+        The FQDN of the Certificate Authorty web enrollment website.
+
+        .PARAMETER username
+        The username to authenticate to the Certificate Authorty web enrollment website.
+
+        .PARAMETER password
+        The password to authenticate to the Certificate Authorty web enrollment website.
```

Comment:
```suggestion
        The password to authenticate to the Microsoft Certificate Authority web enrollment service.
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53